test/addons/dlopen-ping-pong/test.js

```diff
@@ -12,8 +12,16 @@ const bindingPath = require.resolve(`./build/${common.buildType}/binding`);
 console.log('process.dlopen:', bindingPath);
 process.dlopen(module, bindingPath,
                os.constants.dlopen.RTLD_NOW | os.constants.dlopen.RTLD_GLOBAL);
-console.log('module.exports.load:', `${path.dirname(bindingPath)}/ping.so`);
-module.exports.load(`${path.dirname(bindingPath)}/ping.so`);
+
+let pingSOPath = `${path.dirname(bindingPath)}/lib.target/ping.so`;
+
+if (common.isOSX) {
+  pingSOPath = `${path.dirname(bindingPath)}/ping.so`;
+}
+
+console.log('module.exports.load:', pingSOPath);
+module.exports.load(pingSOPath);
+
```

Comment:
@nodejs/build @nodejs/node-gyp this should fix this test based on the changed behavior of gyp. Specifically that non OSX platforms output the ping.so to the `lib/target` directory. 🤞 this fixes CI.

I think there should be a longer conversation if we want this behavior to land in gyp, but we can perhaps push this out in Node.js 15 and get ecosystem feedback and iterate if neccessary.

Please let me know if you have any concerns